src/components/shared/CategoryFilter.tsx

```diff
@@ -0,0 +1,108 @@
+import { BsChevronCompactLeft, BsChevronCompactRight } from 'react-icons/bs';
+import { v4 as uuidv4 } from 'uuid';
+import { ProductAPIDataProps } from '../../utils/dummyCategories';
+
+type categoriesProps = {
+  categories: string[];
+  apiData: ProductAPIDataProps;
+};
+
+function CategoryFilter({ categories, apiData }: categoriesProps) {
+  const [activeItem, setActiveItem] = useState(0);
+
+  let result = apiData.filter(
+    (curProduct) => curProduct.category === categories[0],
+  );
+
+  const [data, setData] = useState(result);
+
+  const filterResult = (category: string) => {
+    result = apiData.filter((curProduct) => curProduct.category === category);
+    setData(result);
+  };
+
+  // Category Onclick event handler for mobile devices
+  const handlePrev = () => {
+    const isFirstCategory = activeItem === 0;
+    const newIndex = isFirstCategory ? categories.length - 1 : activeItem - 1;
+    setActiveItem(newIndex);
+
+    result = apiData.filter(
+      (curProduct) => curProduct.category === categories[newIndex],
+    );
+    setData(result);
+  };
+
+  const handleNext = () => {
+    const isLastSlide = activeItem === categories.length - 1;
+    const newIndex = isLastSlide ? 0 : activeItem + 1;
+    setActiveItem(newIndex);
+
+    result = apiData.filter(
+      (curProduct) => curProduct.category === categories[newIndex],
+    );
+    setData(result);
+  };
+
+  return (
+    <div className="flex flex-col justify-center items-center">
+      <div className="relative hidden sm:flex flex-col sm:flex-row justify-between bg-white shadow p-1 border-2 rounded-lg border-gray-500">
+        {categories.map((item, index) => (
+          <button
+            type="button"
+            key={uuidv4()}
+            className={`px-8 py-1 cursor-pointer ${
+              activeItem === index
+                ? 'border-2 rounded-lg border-gray-500'
+                : 'text-gray-400'
+            }`}
+            onClick={() => {
+              setActiveItem(index);
+              filterResult(categories[index]);
+            }}
+          >
+            {item}
+          </button>
+        ))}
+      </div>
+
+      {/* For small screens */}
+      <div className="flex sm:hidden w-200px justify-between">
+        <button type="button" onClick={handlePrev}>
+          <BsChevronCompactLeft size={24} className="dark:text-white" />
+        </button>
+        <div className="border-2 px-12 py-1 rounded-lg border-gray-500 bg-white">
+          {categories[activeItem]}
+        </div>
+        <button type="button" onClick={handleNext}>
+          <BsChevronCompactRight size={24} className="dark:text-white" />
+        </button>
+      </div>
+      {/* End For small screens */}
+
+      {/* display of dummy api data */}
+      <div className="flex sm:flex-row flex-col justify-between flex-wrap">
+        {data.map((product) => (
+          <div
+            key={product.id}
+            className="w-[220px] dark:text-white px-2 py-2"
```

Comment:
@ISHIMWE-Aime 

I would advise against setting static dimensions (widths and heights) for a component such as this.

Imagine this: You have some space that you want to fill up with something. If you hard-code the dimensions, it might behave terribly against the parent component where it will be used. Because I believe that parent div or component should be the one to set and define the space which a child component is allowed to take.

Use something like `w-full` in each place you used hard-coded dimensions, and allow thee parent where this component will be imported, to be the one to decide how much space it should take.

I recommended the same thing for @SergeInGithub , and it worked, so if you run into some trouble of if you need further demonstration, you can ping us on slack. 

Cheers.